### PR TITLE
feat: add /applications/:app_id endpoint

### PR DESCRIPTION
**Pull request by MatejVukosav**
Add new endpoint to fetch app details

### CODE REVIEW

File: crates/server/src/admin/service.rs [hunk 1]
```diff
@@ -99,7 +105,7 @@ pub(crate) fn setup(
             "/identity/keys",
             delete(handlers::root_keys::delete_auth_keys_handler),
         )
-        .layer(middleware::auth::AuthSignatureLayer::new(store))
+        // .layer(middleware::auth::AuthSignatureLayer::new(store))
```
Comment (miraclx) [inline crates/server/src/admin/service.rs]: ^

File: crates/server/src/admin/service.rs [hunk 2]
```diff
@@ -262,6 +268,43 @@ async fn list_applications_handler(
     }
 }
 
+async fn get_application_details_handler(
+    Path(app_id): Path<String>,
+    Extension(state): Extension<Arc<AdminState>>,
+) -> impl IntoResponse {
+    let app_id_result = match ApplicationId::from_str(&app_id) {
```
Comment (miraclx) [inline crates/server/src/admin/service.rs]: PS: you can write this as this:

```suggestion
    let app_id_result = match app_id.parse() {
```

or if you really want to be explicit about the type definition (though it's inferred correctly from it's use in `get_application` below):

```suggestion
    let app_id_result = match app_id.parse::<ApplicationId>() {
```

**Review by miraclx — APPROVED** [review]

File: crates/server/src/admin/service.rs [hunk 3]
```diff
@@ -262,6 +268,43 @@ async fn list_applications_handler(
     }
 }
 
+async fn get_application_details_handler(
```
Comment (alenmestrov) [inline crates/server/src/admin/service.rs:270]: shouldn't this be in src/admin/handlers/applications.rs?

Comment (MatejVukosav) [reply crates/server/src/admin/service.rs:270]: Fixed https://github.com/calimero-network/core/pull/626

**Pull request merged**